### PR TITLE
LPS-176161 | master

### PR DESCRIPTION
**Pull request by melodywu**
https://issues.liferay.com/browse/LPS-176161

### CODE REVIEW

**Comment by melodywu** [comment]
ci:test:sf

**Comment by melodywu** [comment]
ci:test:sf

**Comment by melodywu** [comment]
ci:test

**Comment by melodywu** [comment]
ci:test:sf

**Comment by melodywu** [comment]
ci:test

**Comment by melodywu** [comment]
ci:test

**Comment by melodywu** [comment]
ci:test

**Comment by melodywu** [comment]
ci:test

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+			CookiePreferenceHandlingAdmin.enablePreferenceHandlingWithCookieConsentInSiteSettings(
+				baseURL = "http://www.able.com:8080",
+				portlet = "Preference Handling",
+				siteURLKey = "guest");
+		}
+
+		task ("When: End User first visits the site") {
+			UserBar.signOut();
+
+			User.firstLoginUI(
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:385]: I think 2 user and first login by the second user is not needed, we could simulate a private window by 
Deleting all existing cookies by CookieHelper.deleteAllCookiesAndRefreshPage(); macro
And asserting the TC's expectation.
What do you think?
cc: @JoyceWang08211

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+				password = "test",
+				specificURL = "http://www.able.com:8080",
+				userEmailAddress = "userea2@liferay.com");
+		}
+
+		task ("Then: Verify only the functional cookie is set cookie banner is present.") {
+			CookieBanner.verifyCookieBannerIsPresent();
+
+			CookieConsentPanel.verifyCookiesAreNotPresent();
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:145]: This macro might need a revisit, to make sure that we are checking that the cookie does not even exist.
Previously we checked if the cookie exists but it has false as value.
@JoyceWang08211

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+
+			CookieHelper._checkSelectedCookieValue(
+				cookieName = "GUEST_LANGUAGE_ID",
+				expectedCookieValue = "en_US");
+		}
+	}
+
+	@description = "This is a use case for LPS-170904 TC-4. Verify only functional cookies are set in Site level enable then disable, when the user isn't a first-time visitor."
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:153]: @melodywu Could you please simplify this description?

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+			User.firstLoginUI(
+				password = "test",
+				specificURL = "http://www.able.com:8080",
+				userEmailAddress = "userea2@liferay.com");
+		}
+
+		task ("Then: Verify only the functional cookie is set cookie banner is present.") {
+			CookieBanner.verifyCookieBannerIsNotPresent();
+
+			CookieConsentPanel.verifyCookiesAreNotPresent();
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:237]: We might had to rename this to
- verifyCookiesForCookieCategoriesAreNotPresent();

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase [hunk 5]
```diff
@@ -81,6 +82,170 @@ definition {
...
+				specificURL = "http://localhost:8080",
+				userEmailAddress = "userea@liferay.com");
+		}
+
+		task ("Then: Verify No cookies are set, not even the functional cookie: GUEST_LANGUAGE_ID verify cookie banner is present.") {
+			CookieBanner.verifyCookieBannerIsPresent();
+
+			CookieConsentPanel.verifyCookiesAreNotPresent();
+
+			CookieHelper._checkSelectedCookieValue(
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase:179]: @melodywu We should extend this macro with the cookies' domain check also. In this case, it's important to know that the cookie's domain is matching with the Instance's domain eg.: localhost

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesConsentPanel.testcase [hunk 6]
```diff
@@ -2,7 +2,8 @@
 definition {
 
 	property ci.retries.disabled = "true";
-	property custom.properties = "feature.flag.LPS-142518=true";
+	property custom.properties = "feature.flag.LPS-142518=true${line.separator}jsonws.web.service.paths.excludes=";
```
Comment (gaborlovasdotcom) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesConsentPanel.testcase:5]: @melodywu I think you wanted to add this custom.poperty to the other test case file?

**Review by gaborlovasdotcom — COMMENTED** [review]
@melodywu I reviewed and added my comments above!
cc: @JoyceWang08211

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase (hunk 1 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:385]: done

Comment (JoyceWang08211) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:385]: I think we could login the site host(aka http://www.able.com:8080) with instance admin(aka test test) and enable explicit cookie consent mode. Then first login site host with the new user. That means adding only one new user is good enough.

Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:385]: done

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase (hunk 2 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:145]: I write macro CookieConsentPanel.verifyCookiesAreNotExist(); thanks and done

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase (hunk 3 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:153]: done

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesConsentPanel.testcase (hunk 6 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesConsentPanel.testcase:5]: done

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase (hunk 4 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesSiteLevelUseCase.testcase:237]: done

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase (hunk 5 above)
Comment (melodywu) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructureee/security/CookiesUseCase.testcase:179]: Yes, I need to extend the macro. Could you teach me how to get domain value from cookie table use javascript? Like CookieHelper._checkSelectedCookieValuemacro 

		```
var javascript = '''var cookieName = "${cookieName}";
                        	var allCookies = document.cookie.split(';');
							var value="empty";
                        	for (var i = 0; i < allCookies.length; i++){
                        		if(allCookies[i].includes(cookieName)){
                        			value = allCookies[i].replace(cookieName+"=","").trim();
                        		}
                        	}
                        	return value;''';
```

**Pull request closed without merging**